src/privileged/testingOverrides/api.js

```diff
@@ -0,0 +1,48 @@
+"use strict";
+
+/* global ExtensionAPI, Preferences */
+
+ChromeUtils.import("resource://gre/modules/Console.jsm");
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+ChromeUtils.import("resource://gre/modules/Preferences.jsm");
+
+this.testingOverrides = class extends ExtensionAPI {
+  getAPI(context) {
+    const { extension } = this;
+
+    // Copied here from tree
+    function makeWidgetId(id) {
+      id = id.toLowerCase();
+      return id.replace(/[^a-z0-9_-]/g, "_");
+    }
+
+    const widgetId = makeWidgetId(extension.manifest.applications.gecko.id);
```

Comment:
Doesn't just extension.id work? Or context.extension.id?

Reply:
So, we didn't write testingOverrides and completely left it in the boilerplate state it was in before. It might be safe to completely remove this, but then again it's only for testing. Since your remarks are mostly nits/questions I think I'd prefer to leave this in the state that it's now in.